rotkehlchen/tasks/manager.py

```diff
@@ -538,6 +541,35 @@ def _maybe_update_snapshot_balances(self) -> Optional[list[gevent.Greenlet]]:
         """
         with self.database.conn.read_ctx() as cursor:
             if self.database.should_save_balances(cursor):
+                detect_tokens: dict[SupportedBlockchain, list[ChecksumEvmAddress]] = defaultdict(list)  # noqa: E501
+                if CachedSettings().get_settings().auto_detect_tokens is True:
+                    type_filter = [event_type.serialize() for event_type in [
```

Comment:
shouldn't you make this smarter? You can never guess all possible combinations and you are 100% definitely missing events here. You should use the direction attribute of the event. IN/OUT/NEUTRAL.